instrumentation/okhttp3/src/main/java/brave/okhttp3/TracingCallFactory.java

```diff
@@ -33,6 +29,12 @@
 // NOTE: this is not an interceptor because the current span can get lost when there's a backlog.
 // This will be completely different after https://github.com/square/okhttp/issues/270
 public final class TracingCallFactory implements Call.Factory {
+  /**
+   * To save overhead, we use a null sentinel when there's no parent. This helps avoid looking at
+   * the current trace context when there was no span in scope at invocation time.
+   */
+  static final TraceContext NULL_SENTINEL = TraceContext.newBuilder()
```

Comment:
note: we only need this because the interceptor can be used two different ways and we can't distinguish based on null alone. This is a conscious alternative to allocating another wrapper.